src/TransponderSquawk.hpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+/**
+ * This class stores a transponder squawk
+ */
+class TransponderSquawk {
+  int16_t value;
```

Comment:
This shouldn't be signed. You made it signed because you needed a magic value for "undefined", but you can use any illegal (positive) value for that, e.g. UINT16_MAX.
Also, for (theoretical) correctness, I'd like new code to use the portable, always-guaranteed-to-be-available types like `uint_least16_t` instead of `uint16_t`, unless an exact size is ultimately necessary (e.g. if binary layout matters, which it does not here).

Reply:
Yes, that was pretty much the "design choice", but you are right. Just using another illegal number (will choose 9) works as well.

Will use the new type as well.